### PR TITLE
Defer fft normalization to post transform stage

### LINKED ISSUE #2050: Delay the normalization of the input when performing fft
This example shows the behavior in python but similar results can be seen in C++ as well.
https://gist.github.com/pavanky/dc64fc83e0fa298e942a3e0ca07ccd15

fft and ifft should be normalized after the transform (not before) for better accuracy.

### PR DESCRIPTION
**Pull request by 9prady9**
Fixes #2050

### CODE REVIEW

**Comment by 9prady9** [comment]
damn, the style fixes commit broke something. The tests are passing with earlier commit. Looking into it.

**Comment by umar456** [comment]
Can you separate the style changes from the PR? Its hard differentiate the changes in functionality from style.

**Comment by 9prady9** [comment]
I made them in two different commits.

**Comment by umar456** [comment]
It looks like you made some style changes in your first commit.

**Comment by 9prady9** [comment]
@umar456 Addressed feedback

File: src/api/c/fftconvolve.cpp [hunk 1]
```diff
@@ -121,63 +134,70 @@ AF_BATCH_KIND identifyBatchKind(const dim4 &sDims, const dim4 &fDims)
...
         AF_BATCH_KIND convBT = identifyBatchKind<baseDim>(sdims, fdims);
 
         ARG_ASSERT(1, (convBT != AF_BATCH_UNSUPPORTED));
 
-        af_array output;
+#define FFTCNV(Ti, Tr, Tc, F0, F1) \
+fftConv<Ti, Tr, Tc, F0 , F1, baseDim>(signal, filter, expand, convBT)
+
+#define FALLBACK(T) \
+fftConvFallback<T , T , T , baseDim>(signal, filter, expand)
```
Comment (umar456) [inline src/api/c/fftconvolve.cpp]: You can do the same thing using some combination of type traits instead of creating macros. You don't need macros here.

**Comment by 9prady9** [comment]
Rolled back style changes for the moment.

**Comment by net147** [comment]
@9prady9 Does the documentation need to be updated? The documentation has norm_factor as scaling the input before the transformation is applied.

**Comment by 9prady9** [comment]
@net147 This doesn't require docs udpate because we didn't normalization in this commit. We merely changed the stage at which normalization is carried out. If a normalization factor other than `1.0` is provided, we use that. The API already has the normalization factor as parameter.

**Comment by net147** [comment]
@9prady9 Yes the API has normalization factor as a parameter, but the documentation says that is factor is applied before the transformation but this commit changes it to be applied after the transformation:
norm_factor "is the normalization factor with which the input is scaled before the transformation is applied"

**Comment by 9prady9** [comment]
Sorry, I didn't notice that it was specific about when it's applied. I will change it, thanks for reporting!

**Comment by 9prady9** [comment]
@net147 I have addressed it in https://github.com/arrayfire/arrayfire/pull/2192 thanks once again

**Pull request merged**